router/schema_validator/object_schema_validator.go

```diff
@@ -140,3 +149,18 @@ func structJsonFields(structType reflect.Type) map[string]reflect.StructField {
 	}
 	return fields
 }
+
+func additionalPropertiesSchema(schema openapi3.Schema) *openapi3.Schema {
+	// if additional properties schema is defined explicitly return it
+	if schema.AdditionalProperties != nil {
+		return schema.AdditionalProperties.Value
```

Comment:
Could the contents of the schema be a Ref?

Reply:
When the spec is parsed, all `Value`s are being populated with what is in `Ref`